src/client/opamCommands.ml

```diff
@@ -4223,6 +4222,7 @@ let clean cli =
            List.iter (fun d ->
                if not (List.mem d keep_sources_dir) then rmdir d))
          switches);
+    if all_switches then cleandir (OpamPath.last_env root);
```

Comment:
I'm unsure if we want to clean last-env in that case. What do you think @dra27 @rjbou?

Reply:
yeah I was hesitating about this because it is not consequent to consider cleaning it.

Reply:
Having discussed it this morning, let's leave the `.last-env` files as _not_ being cleaned:
```suggestion
```
and we'll open an issue to track sorting this out in opam 2.2.1

Reply:
issue created